plugins/catalog-backend-module-bitbucket-server/src/providers/BitbucketServerEntityProvider.ts

```diff
@@ -213,4 +240,332 @@ export class BitbucketServerEntityProvider implements EntityProvider {
+   *
+   * // locationEntities:
+   * // [
+   * //   {
+   * //     kind: 'Location',
+   * //     metadata: {
+   * //       name: 'my-repo',
+   * //       namespace: 'my-project',
+   * //       annotations: {
+   * //         'backstage.io/managed-by-location': 'url/catalog-info-path',
+   * //         'backstage.io/managed-by-origin-location': 'url/catalog-info-path',
+   * //         'host/repo-url': 'url',
+   * //       },
+   * //     },
+   * //     spec: {
+   * //       type: 'bitbucket',
+   * //       target: 'url/catalog-info-path',
+   * //       presence: 'optional',
+   * //     },
+   * //   },
+   * // ]
+   */
+  private async getLocationEntity(event: Events.PushEvent): Promise<Entity[]> {
+    const client = BitbucketServerClient.fromConfig({
+      config: this.integration.config,
+    });
+    const result: Entity[] = [];
+    try {
+      const repository = await client.getRepository({
+        projectKey: event.repository.project.key,
+        repo: event.repository.slug,
+      });
+
+      for await (const entity of this.parser({
+        client,
+        logger: this.logger,
+        location: {
+          type: 'url',
+          target: `${repository.links.self[0].href}${this.config.catalogPath}`,
+          presence: 'optional',
+        },
+      })) {
+        entity.metadata.annotations![
+          this.TARGET_ANNOTATION
+        ] = `${repository.links.self[0].href}${this.config.catalogPath}`;
+        result.push(entity);
+      }
+    } catch (error: any) {
+      if (error.name === 'NotAllowedError') {
+        this.logger.error(error.message);
+      }
+    }
+
+    return result;
+  }
+
+  /** {@inheritdoc @backstage/plugin-events-node#EventSubscriber.supportsEventTopics} */
+  supportsEventTopics(): string[] {
+    return [TOPIC_REPO_PUSH];
+  }
+
+  /** {@inheritdoc @backstage/plugin-events-node#EventSubscriber.onEvent} */
+  async onEvent(params: EventParams): Promise<void> {
+    if (params.topic === TOPIC_REPO_PUSH) {
+      const payload = params.eventPayload as Events.PushEvent;
+
+      if (
+        payload.repository.slug === undefined ||
+        payload.repository.project.key === undefined ||
+        payload.changes === undefined
+      ) {
+        this.logger.error(`Invalid event payload for ${params.topic}.`);
+        return;
+      }
+
+      await this.onRepoPush(params.eventPayload as Events.PushEvent);
+    }
+  }
+
+  /**
+   * Finds if there are existing location entities for the repository that was pushed. If there are, it simply refreshes those entities,
+   * if not, it discovers any entity that was added and removed in the list of entities
+   * @param event - A Bitbucket Server webhook event for repo:refs_change
+   */
+  async onRepoPush(event: Events.PushEvent): Promise<void> {
+    if (!this.canHandleEvents()) {
+      return;
+    }
+
+    if (!this.connection) {
+      throw new Error('Not initialized');
+    }
+
+    if (!this.isHeadPush(event)) return;
```

Comment:
seems to me a bit inconsistent code style (see the two blocks above it).

Reply:
Is this still referring to the method name "isHeadPush"?

Reply:
Sorry, I am still a little confused as to how I should adjust this so the code style matches between the if blocks. Is it the fact that it simply returns with no message instead of throwing an error? In my next commit, I will have some better logging in this area but I don't understand how the style of the if-blocks don't match. Would you possibly be able to make a suggestion?

Reply:
@davelil4 I think he is referring to `return` on the same line and no surrounding curly braces.

Reply:
@pjungermann as opposed to commenting on code style in PR review, perhaps a better solution is to propose an update to the backstage eslint config?

https://eslint.org/docs/latest/rules/curly